webapp/components/notifications/Notification/Notification.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <ds-space :class="[{ read: notification.read }, notification]" margin-bottom="x-small">
+    <no-ssr>
+      <ds-space margin-bottom="x-small">
+        <hc-user
+          v-if="resourceType == 'Post'"
+          :user="post.author"
+          :date-time="post.createdAt"
+          :trunc="35"
+        />
+        <hc-user v-else :user="comment.author" :date-time="comment.createdAt" :trunc="35" />
+      </ds-space>
+      <ds-text color="soft">
+        {{ $t('notifications.menu.mentioned', { resource: resourceType }) }}
+      </ds-text>
+    </no-ssr>
+    <ds-space margin-bottom="x-small" />
+    <nuxt-link
+      class="notification-mention-post"
+      :to="{ name: 'post-id-slug', params, ...hashParam }"
+      @click.native="$emit('read')"
+    >
+      <ds-space margin-bottom="x-small">
+        <ds-card
+          :header="post.title || comment.post.title"
+          hover
+          space="x-small"
+          class="notifications-card"
+        >
+          <ds-space margin-bottom="x-small" />
+          <div v-if="resourceType == 'Post'">{{ post.contentExcerpt | removeHtml }}</div>
+          <div v-else>
+            <span class="comment-notification-header">Comment:</span>
+            {{ comment.contentExcerpt | removeHtml }}
+          </div>
+        </ds-card>
+      </ds-space>
+    </nuxt-link>
+  </ds-space>
+</template>
+
+<script>
+import HcUser from '~/components/User'
+
+export default {
+  name: 'Notification',
+  components: {
+    HcUser,
+  },
+  props: {
+    notification: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    resourceType() {
+      return this.post.id ? 'Post' : 'Comment'
+    },
+    post() {
+      return this.notification.post || {}
+    },
+    comment() {
+      return this.notification.comment || {}
+    },
+    params() {
+      return {
+        id: this.post.id || this.comment.post.id,
+        slug: this.post.slug || this.comment.post.slug,
+      }
+    },
+    hashParam() {
+      return this.post.id ? {} : { hash: `#commentId-${this.comment.id}` }
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-15T23:06:14Z" title="Friday, August 16th 2019, 1:06:14 am +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -42,11 +51,26 @@ export default {
   },
   computed: {
     excerpt() {
-      return this.$filters.removeLinks(this.post.contentExcerpt)
+      const excerpt = this.post.id ? this.post.contentExcerpt : this.comment.contentExcerpt
+      return (
+        (!this.post.id ? '<b>Comment: </b>' : '') + excerpt.replace(/<(?:.|\n)*?>/gm, '').trim()
+      )
     },
     post() {
       return this.notification.post || {}
     },
+    comment() {
+      return this.notification.comment || {}
+    },
+    postParams() {
+      return {
+        id: this.post.id || this.comment.post.id,
+        slug: this.post.slug || this.comment.post.slug,
+      }
+    },
+    hashParam() {
+      return this.post.id ? {} : { hash: `#commentId-${this.comment.id}` }
```


I was thinking about this lately: How about a generic behaviour for fragment identifiers? Can't we simply scroll to *any* html element which has a `scroll-id="blabla"`? We could use it for comments but maybe for other use-cases as well?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-16T08:06:55Z" title="Friday, August 16th 2019, 10:06:55 am +02:00">Aug 16, 2019</time>_

---


You maybe mix up queries with hashtags here? @roschaefer 

A hashtag is always about scrolling to an anchor. Isn’t it?
The naming `commentId` is for the user readability. That they can get a cloud about what the hashtag is, even in case the scrolling does not work for any reason.

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-08-19T14:53:00Z" title="Monday, August 19th 2019, 4:53:00 pm +02:00">Aug 19, 2019</time>_

---


Just read this now – forget my previous comment about the `commentId`, please. I'm all for readability! 😍